src/main/com/fulcrologic/rad/attributes.cljc

```diff
@@ -38,17 +38,21 @@
   If `:ref` is used as the type then the ultimate ID of the target entity should be listed in `m`
   under the ::target key.
   "
-  [kw type m]
-  [qualified-keyword? keyword? map? => ::attribute]
-  (let [v (-> m
-            (assoc ::type type)
-            (assoc ::qualified-key kw))]
-    (when (and (not= :ref type) (or (contains? m ::targets) (contains? m ::target)))
-      (log/warn "NON-Reference attribute" kw "was given referential target(s). This could cause errors in code that generates code from the attribute."))
-    (when (and (= :ref type) (not (contains? m ::targets)) (not (contains? m ::target)))
-      (log/warn "Reference attribute" kw "does not list target(s). Resolver generation will not be accurate."))
-    #?(:clj  (registered-map (or (-> m meta :registration-key) kw) v)
-       :cljs v)))
+  ([kw type m]
+   [qualified-keyword? keyword? map? => ::attribute]
+   (new-attribute kw type nil m))
+  ([kw type docstring m]
```

Comment:
I do not agree with reifying the docstring in the data model. This increases code size, possibly by a lot, and the most useful purpose of a docstring is IDE help, which is taken from the metadata (or notation, depending on the coding system) of the macro.

Reply:
mm... What do you think about making the docstr capture only apply to the CLJ build (elided entirely in the cljs build)? I'm using some custom tools to run over the attribute list we compile, and it doesn't use the raw code, so I need a way to capture the docstr into the attribute map at least sometimes.

Reply:
If you want text in the model, why not make that a formal option on the attribute? A docstring is meant to be a developer-only comment, not part of the data model. There are lots of different possible strings on the data model: API documentation string, labels for different UI use-cases, Info popups for UI, etc.

Reply:
Because it truly is a docstring. I've built some internal tooling to viz our data model, and I'm showing the docstring on hover for additional context. 

<img width="892" alt="image" src="https://github.com/fulcrologic/fulcro-rad/assets/16200906/2e997d39-8dd7-4c97-bdf2-355d20c14e03">

Reply:
I see. But your tooling can grab the metadata off the var can't it?

Reply:
Possibly. I could change it to iterate over the vars of `all-attributes`. e.g. 

```clojure
(def attributes [#'id #'capacity ...])
```

But then this is a more involved change.